File: algoS_p2/dijkstra/Graph.py
#!/usr/bin/env python
from Vertex import Vertex

class Graph:
    def __init__(self):
        self.vertices = {}
        self.E = 0

    def addEdge(self, start, end, weight):
        try:
            if start not in self.vertices:
                v = Vertex(start)
                self.vertices[start] = v
            if end not in self.vertices:
                v = Vertex(end)
                self.vertices[end] = v
            self.vertices[start].adj[end] = weight
            self.E += 1
        except:
            print("Error in addEdge()")

    def printVertices(self):
        for v in self.vertices:
            print(v, ' v.d: ', self.vertices[v].d)

File: algoS_p3/greedy/Graph.cpp
#include "Graph.h"
#include <iostream>
#include <stack>
#include <typeinfo>


Graph::Graph(int V) {
    vertices = std::vector<Vertex>(V);
}

void Graph::addEdge(int start, int end, int weight) {
    try {
        if (vertices[start].val == -1) vertices[start].val = start;
        if (vertices[end].val == -1) vertices[end].val = end;
        vertices[start].adj[end] = weight;
        vertices[end].adj[start] = weight; // remember to udpate end-start edge when operating on undirected graphs
    } catch (...) {
        std::cout << "in catch" << std::endl;
    }
   
}

std::unordered_map<int, int>& Graph::getAdj(int v) {
    return vertices.at(v).adj;
}


